rholang/src/main/scala/coop/rchain/rholang/interpreter/RhoHistoryRepositorySyntax.scala

```diff
@@ -0,0 +1,30 @@
+package coop.rchain.rholang.interpreter
+
+import cats.effect.Sync
+import cats.syntax.all._
+import coop.rchain.rholang.interpreter.RhoRuntime.RhoHistoryRepository
+import coop.rchain.rholang.interpreter.merging.RholangMergingLogic.convertToReadNumber
+import coop.rchain.rspace.hashing.Blake2b256Hash
+import coop.rchain.rspace.syntax._
+
+trait RhoHistoryRepositorySyntax {
+  implicit final def rhoHistoryRepositoryRholangSyntax[F[_]](
+      hr: RhoHistoryRepository[F]
+  ): RhoHistoryRepositoryOps[F] = new RhoHistoryRepositoryOps(hr)
+}
+
+final class RhoHistoryRepositoryOps[F[_]](private val hr: RhoHistoryRepository[F]) extends AnyVal {
+  def readMergeableValues(
+      baseState: Blake2b256Hash,
+      channelHashes: Set[Blake2b256Hash]
+  )(implicit sync: Sync[F]): F[Map[Blake2b256Hash, Long]] =
+    for {
+      historyReader <- hr.getHistoryReader(baseState)
+      baseReader    = historyReader.readerBinary
```

Comment:
Why to create extension for RhoHistoryRepository when you need only HistoryReaderBinary?

Reply:
```
def convertToReadNumber[F[_]: Monad](
      getDataFunc: Blake2b256Hash => F[Seq[Datum[ListParWithRandom]]]
  ): Blake2b256Hash => F[Option[Long]] =
```
requires specific types, and HistoryReaderBinary is [F, C, P, A, K]. I'm not sure what's that good way to have it on a reader.

Reply:
Extension functions should be defined for stuff used often, but if it's hard to define it or not used on many places, maybe it's not the right candidate to be extension. It can be just a regular helper function.

Reply:
Do you want me to remove extension?